assets/js/view/part/revenue.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+require 'view/part/base'
+require 'view/part/multi_revenue'
+
+module View
+  module Part
+    class Revenue < Base
+      needs :tile
+
+      def preferred_render_locations
+        if multi_revenue?
+          return [
+            {
+              regions: %w[half_edge5 half_edge0 half_edge1],
+              transform: 'translate(-30 20)',
+            },
+            {
+              regions: %w[half_edge2 half_edge3 half_edge4],
+              transform: 'translate(-30 -20)',
+            },
+          ]
+        end
+
+        case @slots
+        when 1
+          [
+            {
+              regions: ['half_corner1.5', 'half_edge1', 'half_edge2'],
+              transform: 'translate(-45 0)',
+            },
+            {
+              regions: ['half_corner4.5', 'half_edge4', 'half_edge5'],
+              transform: 'translate(45 0)',
+            },
+            {
+              regions: %w[half_edge1 edge1],
+              transform: 'translate(-45 25)',
+            },
+            {
+              regions: ['corner1.5'],
+              transform: 'translate(-65 0)',
+            },
+            {
+              regions: ['corner4.5'],
+              transform: 'translate(65 0)',
+            },
+          ]
+        when (2..4)
+          [
+            {
+              regions: ['corner1.5'],
+              transform: 'translate(-70 0)',
+            },
+            {
+              regions: ['corner4.5'],
+              transform: 'translate(70 0)',
+            },
+            {
+              regions: ['half_edge1'],
+              transform: 'translate(-45 25)',
+            },
+          ]
+        else
+          [
+            {
+              regions: ['center'],
+              transform: 'translate(0 0)',
+            }
+          ]
+        end
+      end
+
+      def parse_tile
+        @slots = @tile.cities.map(&:slots).sum + @tile.towns.size
+
+        revenue_stops = @tile.cities + @tile.towns + @tile.offboards
+        revenues = revenue_stops.map(&:revenue).uniq
+
+        return if revenues.empty?
+
+        if revenues.size == 1
+          revenues = revenues.first
+        else
+          puts 'WARNING: encountered multiple different revenues on one tile'
+          @revenue = nil
+          return
+        end
+
+        @revenue =
+          if revenues.values.uniq.size == 1
```

Comment:
`revenues` starts as a list of hashes. The `revenues = revenues.first` bit sets `revenues` to a hash.

It starts as a list since there can be multiple stops on one tile. Usually they generate the same revenue but I did find some tiles where the different stops generate different revenues. I'm willing to admit worrying about such cases now is too much future-proofing :slightly_smiling_face: